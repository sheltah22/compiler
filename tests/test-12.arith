let power : int -> (int -> int) =
fix f (x:int) : (int -> int) ->
fun (y:int) : int ->
  if y = 0
  then 1
  else x * ((f x) (y - 1))
in
  (power 6) 3
