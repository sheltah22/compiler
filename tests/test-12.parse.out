(let power : int -> int -> int = (fix f (x: int) : int -> int -> (fun (y: int) : int -> (if (= y 0) 1 (* x ((f x) (- y 1)))))) in ((power 6) 3))
